pallets/creator-staking/src/functions.rs

```diff
@@ -0,0 +1,309 @@
+        ensure!(era < current_era, Error::<T>::EraOutOfBounds);
+
+        ensure!(
+                !creator_stake_info.creator_reward_claimed,
+                Error::<T>::AlreadyClaimedInThisEra,
+            );
+        ensure!(
+                creator_stake_info.total > Zero::zero(),
+                Error::<T>::NotStakedCreator,
+            );
+
+        let rewards_and_stakes =
+            Self::general_era_info(era).ok_or(Error::<T>::UnknownEraReward)?;
+
+        // Calculate the creator reward for this era.
+        let (creator_reward, _) = Self::distributed_rewards_between_creator_and_stakers(creator_stake_info, &rewards_and_stakes);
+
+        Ok(creator_reward)
+    }
+
+    /// `Err` if pallet disabled for maintenance, `Ok` otherwise
+    pub fn ensure_pallet_enabled() -> Result<(), Error<T>> {
+        if PalletDisabled::<T>::get() {
+            Err(Error::<T>::PalletIsDisabled)
+        } else {
+            Ok(())
+        }
+    }
+
+    /// Returns available staking balance for the potential staker
+    pub(super) fn available_staking_balance(
+        staker: &T::AccountId,
+        ledger: &AccountLedger<BalanceOf<T>>,
+    ) -> BalanceOf<T> {
+        // Ensure that staker has enough balance to bond & stake.
+        let free_balance =
+            T::Currency::free_balance(staker).saturating_sub(T::MinimumRemainingAmount::get());
+
+        // Remove already locked funds from the free balance
+        free_balance.saturating_sub(ledger.locked)
+    }
+
+    /// `true` if creator is active, `false` if it has been unregistered
+    pub(super) fn is_creator_active(space_id: SpaceId) -> bool {
+        RegisteredCreators::<T>::get(space_id)
+            .map_or(false, |info| info.state == CreatorState::Registered)
+    }
+
+    /// `true` if all the conditions for restaking the reward have been met, `false` otherwise
+    pub(crate) fn should_restake_reward(
+        restake: bool,
+        creator_state: CreatorState,
+        latest_staked_value: BalanceOf<T>,
+    ) -> bool {
+        restake
+            && creator_state == CreatorState::Registered
+            && latest_staked_value > Zero::zero()
+    }
+
+    pub(super) fn do_unregister_creator(
+        space_id: SpaceId,
+        unregister_origin: UnregistrationAuthority<T::AccountId>,
+    ) -> DispatchResultWithPostInfo {
+        let mut creator_info =
+            RegisteredCreators::<T>::get(space_id).ok_or(Error::<T>::NotRegisteredCreator)?;
+
+        ensure!(creator_info.state == CreatorState::Registered, Error::<T>::NotRegisteredCreator);
+        let stakeholder = creator_info.stakeholder.clone();
+
+        // TODO: make flexible register deposit
+        if let UnregistrationAuthority::Root = unregister_origin {
+            T::Currency::slash_reserved(&stakeholder, T::RegistrationDeposit::get());
+        } else if let UnregistrationAuthority::Creator(who) = unregister_origin {
+            ensure!(who == stakeholder, Error::<T>::NotACreator);
```

Comment:
It is much more logical to check that only CURRENT space owner can unregister their space from the creators registry.
```suggestion
            ensure!(who == space.owner, Error::<T>::NotACreator);
```